specs/0109-infobar-component.md

```diff
@@ -0,0 +1,71 @@
+* Feature Name: Infobar Component
+* Created: 2021-02-25
+* Associated Bugs: [bug 1694911](https://bugzilla.mozilla.org/show_bug.cgi?id=1694911)
+
+# Summary
+
+Create a reusable component to display infobars to users.
+
+# Motivation
+
+We need a flexible way to notify users without writing ad-hoc code for every situation:
+* Infobars should be manageable by admins without intervention from developers.
+* Unlike notifications, infobars are more visible and require interaction from users to get dismissed.
+
+Example use cases:
+* We introduce a brand new feature, like string comments, and we want to make sure that users are aware of it. The infobar will include a short message and a link to documentation.
+* We want to warn users of an upcoming maintenance window for Pontoon.
+
+# Feature explanation
+
+When a logged in user opens Pontoon, we check against a list of infobars. If the user matches the necessary conditions, we display the infobar.
+
+Information about each infobar is stored in a table with the following fields:
+* Identifier: short text identifier, e.g. `new_terms_2020`.
+* Start time (datetime): when we should start displaying the infobar to users.
+* End time (datetime): when we should stop displaying the infobar to users.
+* Creation date (date): stores when the infobar was created.
+* Cohort: if defined, we only display the infobar for users part of a specific experiment or cohort (A/B testing).
+* Selector (optional): a CSS selector.
+    * If empty, the infobar will be displayed at the top of the page.
+    * If defined, the infobar will be displayed as a pop-up above the element identified by the selector.
+* Pages pattern (optional): regular expression to indicate on which pages the infobar should be displayed.
+    * If empty, the infobar will be displayed on all pages.
+    * If defined, the infobar will be displayed only on pages whose URL matches the regular expression.
+* Title (optional): title for the infobar.
+* Text (mandatory): text for the infobar.
+    * Basic HTML should be allowed, including anchors.
```

Comment:
This should be defined, as we need to instruct bleach on what to accept. Do we enforce a link target?

Reply:
Listing `<p>`, `<strong>`, `<em>`, `<a>`, I don't think we'd need more.